### DISCUSSION TITLE
Short animation in documentation

### DISCUSSION THREAD

**Discussion by PaulWessel** [GenericMappingTools/gmt#5693] [Documentation development discussion]
I would like to greatly improve the documentation for **events** and **movie** by adding more examples on how to animate symbols, lines, and polygons and short (say 24 frames) animations that go with that.  Since RST and browsers support both animated GIFs and MP4s, I think for these cases the best effect would be a looped GIF.  So the questions are

1. Do I implement the required animations as documentation scripts (in doc/scripts) like all (?) other examples and scripts used to display something in the RST?  If so, I probably need to use **-x**1 to not compete with the ctest parallelization.
2. Or, do I render the small GIFs and stick them in with the static precomputed images in docs/fig (or possibly a new docs/animation)?

Unlike the grandiose HD animations that we place on YouTube, these are tiny things, such as this one below which is nevertheless 94K:

![trajectory3](https://user-images.githubusercontent.com/26473567/130337975-763082e9-203d-4bb3-906d-50a64f194f50.gif)

Let me know how think I should do this, @GenericMappingTools/core.

**Comment by maxrjones** [comment]
I agree that this would be a great improvement to the docs.

My preference would be to decide and implement a medium to long-term solution for postscript and binary/large files (see https://github.com/GenericMappingTools/gmt/issues/3470, https://github.com/GenericMappingTools/gmt/issues/2681) before embarking on these improvements, which fit within the broader goals of adding images throughout the documentation (e.g., https://github.com/GenericMappingTools/gmt/issues/4913, https://github.com/GenericMappingTools/gmt/issues/4379, https://github.com/GenericMappingTools/gmt/issues/3898) and adding how-to guides/smaller examples (https://github.com/GenericMappingTools/gmt/pull/5364). While linking to GitHub-hosted images has been used as a temporary solution, I consider this a technical debt since it may lead to more work later on when tracking/metadata about the figure is needed. I have on the agenda for the next community meeting a discussion about one possible solution (data version control, DVC). This has worked well for PyGMT, but I would need to do more research about how this could work for building GMT docs.
